app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -73,31 +76,34 @@ class FxaRepo(
         return accountManager.beginAuthenticationAsync()
     }
 
-    private inner class FirefoxAccountObserver : AccountObserver {
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    inner class FirefoxAccountObserver : AccountObserver {
+        /**
+         * The account profile is fetched asynchronously.
+         * There is no way to transition from [NOT_AUTHENTICATED] directly to [AUTHENTICATED_WITH_PROFILE];
+         * [AUTHENTICATED_NO_PROFILE] is always a state in between.
+         */
         override fun onAuthenticated(account: OAuthAccount) {
             logger.debug("onAuthenticated")
```

Comment:
nit: I'd like to remove these eventually so if they're not useful now, feel free to remove them.